### PR TITLE
README: Update releases

### PR DESCRIPTION
**Pull request by gentoo-root**

### CODE REVIEW

**Review by learnitall — APPROVED** [review]

File: README.rst [hunk 1]
```diff
@@ -80,7 +80,7 @@ corresponding image pull tags and their release notes where applicable:
...
 | `main <https://github.com/cilium/cilium/commits/main>`__                             | daily      | ``quay.io/cilium/cilium-ci:latest``          | N/A                                                                                       |
 +--------------------------------------------------------------------------------------+------------+----------------------------------------------+-------------------------------------------------------------------------------------------+
-| `v1.15.0-rc.1 <https://github.com/cilium/cilium/commits/v1.15.0-rc.1>`__             | 2024-01-16 | ``quay.io/cilium/cilium:v1.15.0-rc.1``       | `Release Candidate Notes <https://github.com/cilium/cilium/releases/tag/v1.15.0-rc.1>`__  |
+| `v1.15.0-pre.3 <https://github.com/cilium/cilium/commits/v1.15.0-pre.3>`__           | 2023-12-04 | ``quay.io/cilium/cilium:v1.15.0-pre.3``      | `Release Candidate Notes <https://github.com/cilium/cilium/releases/tag/v1.15.0-pre.3>`__ |
```
Comment (bimmlerd) [inline README.rst]: Do we really want this change?

**Review by bimmlerd — CHANGES_REQUESTED** [review]

File: README.rst (hunk 1 above)
Comment (gentoo-root) [reply README.rst]: I don't know, it was added by the bump-readme.sh script. Please advise.

Comment (qmonnet) [reply README.rst]: I'd say let's keep `v1.15.0-rc-1`, and we should look into fixing the script as a follow-up

**Review by qmonnet — APPROVED** [review]
Looks good apart from David's comment, thanks!

File: README.rst (hunk 1 above)
Comment (qmonnet) [reply README.rst]: The script fails because it picks the latest tag on the branch, but `v1.15.0-rc-1` is on `v1.15`, not on `main`.

**Review by qmonnet — APPROVED** [review]

**Review by bimmlerd — APPROVED** [review]
thanks!

File: README.rst (hunk 1 above)
Comment (qmonnet) [reply README.rst]: https://github.com/cilium/cilium/pull/30412

**Pull request merged**